sdks/go/pkg/beam/core/util/reflectx/calls.tmpl

```diff
@@ -57,7 +60,9 @@ func (c *shimFunc{{$in}}x{{$out}}) Call{{$in}}x{{$out}}({{mkargs $in "arg%v" "in
 
 func ToFunc{{$in}}x{{$out}}(c Func) Func{{$in}}x{{$out}} {
     if c.Type().NumIn() != {{$in}} || c.Type().NumOut() != {{$out}} {
-        panic("incompatible func type")
+        panic(fmt.Sprintf("incompatible func type, expected function of {{$in}} " +
+	"inputs & {{$out}} outputs and instead received a function of %d inputs and %d" +
```

Comment:
Good change.

Small suggestion: maybe include the full function type as well? Btw, the usual way of writing these error messages in the opinionated language of Go is a terse got-then-want style: "blah: %v, want %v".